pages/checkout.vue

```diff
@@ -409,18 +369,130 @@ export default {
             },
           },
           isDestinationCommercial: false,
-          shippingMethodCode: "",
-          shippingMethodName: shippingDetails.value.shippingMethod,
+          shippingMethodCode: shippingDetails.value.shippingMethod.code,
+          shippingMethodName: shippingDetails.value.shippingMethod.name,
         },
       }
 
       await setShippingInfo(params)
+      populateShippingDetails()
+
+      await loadShippingMethods(checkout.value.id)
+      populateShppingMethodDetails()
+    }
+
+    // shippingMethods
+    const shippingMethodDetails = ref({
+      shipItems: [],
+      pickupItems: [],
+      deliveryItems: [],
+      shippingRates: [],
+    })
+
+    const populateShppingMethodDetails = () => {
+      shippingMethodDetails.value = {
+        shipItems: checkoutGetters.getShipItems(checkout.value),
+        pickupItems: checkoutGetters.getPickupItems(checkout.value),
+        deliveryItems: checkoutGetters.getDeliveryItems(checkout.value),
+      }
+
+      shippingMethodDetails.value.shipItems = shippingMethodDetails.value.shipItems.map((item) => ({
+        ...item,
+        shippingRates: shippingMethodGetters.getShippingRates(shippingMethods.value),
+      }))
+
+      console.log("---shippingMethodDetails.value-----", shippingMethodDetails.value)
+    }
+
+    // billing
+    const billingDetails = ref({
+      firstName: "",
+      lastName: "",
+      streetName: "",
+      apartment: "",
+      city: "",
+      state: "",
+      zipCode: "",
+      country: "",
+      phoneNumber: "",
+    })
+
+    const populateBillingDetails = () => {
+      billingDetails.value = shopperContactGetters.getBillingDetails(
+        checkout.value?.billingInfo?.billingContact
+      )
+    }
+
+    const updateBillingDetails = (newBillingDetails) => {
+      billingDetails.value = { ...newBillingDetails }
     }
 
+    const saveBillingDetails = async () => {
+      const params = {
```

Comment:
This Input building should be in a separate utility function
Note: if you remove the "streetName" remapping and keep data similar to the API structure, this becomes a lot simpler